src/components/legend.ts

```diff
@@ -326,6 +329,38 @@ export module Components {
       }
     }
 
+    /**
+     * Gets the function determining the opacity of the symbols of the Legend.
+     *
+     * @returns {(datum: any, index: number) => number}
+     */
+    public symbolOpacity(): (datum: any, index: number) => number;
+    /**
+     * Sets the function determining the symbols of the Legend.
+     *
+     * @param {(datum: any, index: number) => number} opacityFunction
+     * @returns {Legend} The calling Legend
+     */
+    public symbolOpacity(opacityFunction: (datum: any, index: number) => number): Legend;
+    /**
+     * Sets the function determining the symbols of the Legend.
+     *
+     * @param {number} constantOpacity
+     * @returns {Legend} The calling Legend
+     */
+    public symbolOpacity(constantOpacity: number): Legend;
```

Comment:
Feel free to combine the above two setters into a single union type

Let's also call the parameter `symbolOpacity`